src/extension/src/coreTemplateStudio.ts

```diff
@@ -94,144 +84,151 @@ export class CoreTemplateStudio {
     }
   }
 
-  private constructor(process: ChildProcess, port: number, url: string) {
-    this._process = process;
-    this._port = port;
-    this._url = url;
+  private constructor(processCli: ChildProcess) {
+    this._processCli = processCli;
+    this.cliEvents = new CliEventEmitter();
+    this.readStream(this._processCli);
   }
 
-  public getPort(): number {
-    return this._port;
+  // This function is a listener, in the constructor, it gets attached
+  // then it will always get triggered when a command is write to the cli and there is responses from cli, until the process gets killed
+  public async readStream(process: ChildProcess) {
+    let data = "";
+    process.stdout.on("data", (chunk) => {
+      data += chunk;
+      var responses = data.toString().split("\n");
+      for (let i = 0; i < responses.length - 1; i++) {
+        var result = JSON.parse(responses[i]);
+        this.cliEvents.emit(result['messageType'], result['content']);
+      }
+      data = responses[responses.length - 1];
+    });
+
+    process.stderr.on("data", (data) => {
```

Comment:
According to your comment 

> it will always get triggered when a command is write to the cli and there is responses from cli, until the process gets killed

It seems like this function relies on the fact that the child process will *always* write on `stderr` before exiting.

I would recommend using `process.exit` function instead.

Reply:
Well technically, the CLI should *never* exit. Of course, things can go wrong and cause the CLI to crash unexpectedly. Thus, maybe it's better if we handle both cases.

Reply:
Great Catch! 

Will something like this handle both cases?
`    try {
      process.stderr.on("data", (data) => {
        this.cliEvents.emit("error", data.toString());
      });
    } catch (error) {
      process.on("exit", (error) => {
        this.cliEvents.emit("error", error.toString());
      });
    }`

Reply:
According to Node.js documentation https://nodejs.org/api/process.html#process_event_exit, I would rather use something like this:

``` typescript
process.stderr.on("data", (data) => this.cliEvents.emit("error", data.toString()));
process.on("exit", (code) => this.cliEvents.removeAllListeners());
```

Reply:
Updated PR accordingly :)